[REFACTOR] Improve naming and structure clean up Program Main

nadav5 opened the pull request:
Replaced unclear variable names with meaningful ones
Removed 'var'
Extracted business logic from Program.Main into ApplicationRunner

nadav5 committed:
[IMPROVE] PCAP and ICD structure

No masks, just no FLOAT and just UINT instead. With the use of Scale

nadav5 committed:
[REDACTOR] Task and Variables

nadav5 committed:
[REDACTOR] DI

nadav5 committed:
[IMPROVE] DI and structure

nadav5 committed:
[IMPROVE] DI Constant structure

nadav5 committed:
[IMPROVE] Work in bits

nadav5 committed:
[REFACTOR] Project folders and rename classes

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantCsv.cs:
```diff
@@ -0,0 +1,18 @@
+using System.Threading.Tasks;
+
+namespace SendRecieveUDP.Common.Constant
+{
```
Unnecessary use of the 'Common' folder. In .Net projects there is not need for a common folder in solutions with one project inside of them. Instead, move the Constants into the Models folder

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantBits.cs:
```diff
@@ -0,0 +1,16 @@
+        public const int BITS_IN_BYTE = 8;
+        public const int ROUND_TO_BYTE = 7;
+        public const int BYTE_SIZE = 1;
+        public const int ZERO =0 ;
```
Unclear meaning of the constants. Change the names to convey their meaning. For example: ZERO - if it's just a number why create a constant? and if the constant represents something then it needs to be conveyed in the name

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantCsv.cs:
```diff
@@ -0,0 +1,18 @@
+        public const int ONLY_TITLES = 2;
+        public const int EMPTY = 0;
+        public const int ONE_LINE = 1;
+        public const string CLUSTER = "c_";
```
Again, unclear meaning of the constants, it needs to be clear what they mean before you look at their usage. For example: ONE_LINE - unless you look at the code and see where you used it it's unclear what this constant represents

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantNetwork.cs:
```diff
@@ -0,0 +1,14 @@
+    public class ConstantNetwork
+    {
+        public const int PORT = 5000;
+        public const string LOCALHOST = "127.0.0.1";
```
This address is called Loopback

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantNetwork.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public class ConstantNetwork
+    {
+        public const int PORT = 5000;
```
This represents a port, but the port of what? kafka? the application itself?

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantTime.cs:
```diff
@@ -0,0 +1,13 @@
+{
+    public class ConstantTime
+    {
+        public const int MINUTE = 60;
```
Change to SECONDS_IN_MINUTE

ShalevTamir commented on SendRecieveUDP/Common/Constant/ConstantTypes.cs:
```diff
@@ -0,0 +1,19 @@
+        public const int INT_SIZE = 4;
+        public const int FLOAT_SIZE = 4;
+        public const ulong ONE_LONG_SIZE = 1UL;
+        public const ulong ZERO_LONG_SIZE = 0UL;
```
Same here, the names don't convey the meaning clearly

ShalevTamir commented on SendRecieveUDP/Model/Interfaces/DataCleansing/ICsvCleaner.cs:
```diff
@@ -0,0 +1,13 @@
+{
+    public interface ICsvCleaner
+    {
+        void Run(string inputFile, string outputFile);
```
The name run doesn't provide much into the work that it's doing, if this is a CsvCleaner then the main method would 'Clean'. Also the name Cleaner is a bit unclear, try to find terminologies that are more code-related, for example: Formatter

ShalevTamir commented on SendRecieveUDP/Model/Interfaces/Networking/IUdpReceiver.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public interface IUdpReceiver
+    {
+        void ReceiveUDP(List<IcdField> icd, CancellationToken token);
```
Perfect, this is an example of the previous review I wrote you, A udp receiver - so it's main method would receive udp. Unlike the CsvCleaner which had run as it's main method

ShalevTamir commented on SendRecieveUDP/Program.cs:
```diff
@@ -1,35 +1,49 @@
+            services.AddScoped<IUdpSender, UdpSender>();
+            services.AddScoped<IUdpReceiver, UdpReceiver>();
+            services.AddScoped<IBitEncoder, BitEncoder>();
+            services.AddScoped<IPacketBuilder, PacketBuilder>();
```
Why are these scoped and not singleton?

ShalevTamir commented on SendRecieveUDP/Program.cs:
```diff
@@ -1,35 +1,49 @@
-            recieverThread.IsBackground = true;
-            recieverThread.Start();
-            Thread.Sleep(500);
+            using var provider = services.BuildServiceProvider();
```
Unnecessary use of var

ShalevTamir commented on SendRecieveUDP/Program.cs:
```diff
@@ -1,35 +1,49 @@
 
 
+            var cancellationToken = new CancellationTokenSource();
+            var reciever = provider.GetRequiredService<IUdpReceiver>();
```
Unecessary use of var

ShalevTamir commented on SendRecieveUDP/Program.cs:
```diff
@@ -1,35 +1,49 @@
 
 
-            Send.SendCSV("Longest_Master_23517_clean.csv", icd);
+            var cleaner = provider.GetRequiredService<ICsvCleaner>();
```
Unecessary use of var

ShalevTamir commented on SendRecieveUDP/Program.cs:
```diff
@@ -1,35 +1,49 @@
+            var cleaner = provider.GetRequiredService<ICsvCleaner>();
+            cleaner.Run("5ROW.csv", "Longest_Master_23517_clean.csv");
 
+            var sender = provider.GetRequiredService<IUdpSender>();
```
Unecessary use of var

ShalevTamir commented on SendRecieveUDP/Program.cs:
```diff
@@ -1,35 +1,49 @@
 {
     internal class Program
     {
         static void Main(string[] args)
```
The main method shouldn't include any logic. There should be one main service that encapsulates your logic and the Main method calls it. Think of it like a controller in .Net Core applications. In the controller you don't implement any logic in the endpoint methods, you use a proxy service that provides the project's logic

ShalevTamir commented on SendRecieveUDP/Service/DataCleansing/CsvCleaner.cs:
```diff
@@ -0,0 +1,47 @@
+        {
+            if (!File.Exists(inputFile))
+            {
+                Console.WriteLine($" File {inputFile} not found!");
```
Don't use Console.WriteLine, instead use Debug.WriteLine. It's more efficient. The best practice is using a logger service. In your case we don't want to create a logger yet so you can create a service that uses Debug.WriteLine, in order to allow scalability and proper logging in the future

ShalevTamir commented on SendRecieveUDP/Service/DataCleansing/CsvCleaner.cs:
```diff
@@ -0,0 +1,47 @@
+            if (!File.Exists(inputFile))
+            {
+                Console.WriteLine($" File {inputFile} not found!");
+                return;
```
Don't use return in void methods, try to think of another way to use validation

ShalevTamir commented on SendRecieveUDP/Service/BitProcessing/BitEncoder.cs:
```diff
@@ -0,0 +1,38 @@
+        {
+            for (int indexInByte = ConstantBits.ZERO; indexInByte < bitCount; indexInByte++)
+            {
+                int byteIdx = (bitOffset + indexInByte) / ConstantBits.BITS_IN_BYTE;
```
Don't use shothands when unnecessary, instead of byteIdx, use byteIndex. In this case there is not reason to shorten it and using the full word makes the meaning more clear

ShalevTamir commented on SendRecieveUDP/Service/DataCleansing/CsvCleaner.cs:
```diff
@@ -0,0 +1,47 @@
+
+            for (int rowIndex = ConstantCsv.ONE_LINE; rowIndex < lines.Length; rowIndex++)
+            {
+                string[] parts = lines[rowIndex].Split(ConstantCsv.CSV_DELIMITER);
```
parts? did you mean columns?

ShalevTamir commented on SendRecieveUDP/Service/DataCleansing/CsvCleaner.cs:
```diff
@@ -0,0 +1,47 @@
+                for (int column = ConstantCsv.EMPTY; column < parts.Length; column++)
+                {
+                    if (parts[column].StartsWith(ConstantCsv.CLUSTER))
+                        parts[column] = parts[column].Substring(ConstantCsv.ONLY_TITLES);
```
Unclear what you are doing here

ShalevTamir commented on SendRecieveUDP/Service/Networking/UdpSender.cs:
```diff
@@ -0,0 +1,47 @@
+            if (lines.Length < ConstantCsv.ONLY_TITLES)
+            {
+                Console.WriteLine("CSV file does not contain enough lines.");
+                return;
```
Again, don't use return on void methods

ShalevTamir commented on SendRecieveUDP/Service/Networking/UdpSender.cs:
```diff
@@ -0,0 +1,47 @@
+            string[] headers = lines[ConstantCsv.TITLE_ROW].Split(ConstantCsv.CSV_DELIMITER);
+            var dataLines = lines.Skip(ConstantCsv.ONE_LINE);
+
+            var headerIndex = headers
```
Unclear meaning of the variable because of var usage

ShalevTamir commented on SendRecieveUDP/Service/Networking/UdpSender.cs:
```diff
@@ -0,0 +1,47 @@
+            }
+
+            string[] headers = lines[ConstantCsv.TITLE_ROW].Split(ConstantCsv.CSV_DELIMITER);
+            var dataLines = lines.Skip(ConstantCsv.ONE_LINE);
```
Unnecessary var usage

ShalevTamir commented on SendRecieveUDP/Service/Packet/PacketBuilder.cs:
```diff
@@ -0,0 +1,87 @@
+            return packet;
+        }
+
+        private void FillPacket(byte[] packet, string[] csvColumns, List<IcdField> icd, Dictionary<string, int> headerIndex)
```
Unclear name meaning

ShalevTamir commented on SendRecieveUDP/Service/Packet/PacketBuilder.cs:
```diff
@@ -0,0 +1,87 @@
+
+namespace SendRecieveUDP.Service.Packet
+{
+    public class PacketBuilder : IPacketBuilder
```
If it's a packet builder how come it builds a packet **and** decodes it? the name should convey the entire responsiblity of the class

ShalevTamir reviewed (changes requested):
A couple of points:
1. Notice the names of classes, variables, constants... give more meaningful names that convey the responsibllity of the object
2. Next time split the merge request into smaller merge requests. notice that this merge request has 25 changes, this is too large for one merge request

nadav5 committed:
[REFACTOR] Delete Var and bring in more meaningful names

nadav5 committed:
[REFACTOR] Instead of return in void function return Function Result

nadav5 committed:
[REFACTOR] Move logic from Program to ApplicationRunner

ShalevTamir commented on SendRecieveUDP/Model/Constant/ConstantBits.cs:
```diff
@@ -0,0 +1,16 @@
+    public class ConstantBits
+    {
+        public const int BITS_IN_BYTE = 8;
+        public const int MAX_BIT_INDEX_IN_BYTE = 7;
```
You can use BITS_IN_BYTE - 1, no need to create a different constant for this

ShalevTamir commented on SendRecieveUDP/Model/Ro/FunctionResult.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace SendRecieveUDP.Model.Ro
+{
+    public class FunctionResult
```
The name is too generic, you created this object to represent the status of something specific, so the name should convey this

ShalevTamir commented on SendRecieveUDP/Service/Application/ApplicationRunner.cs:
```diff
@@ -0,0 +1,46 @@
+
+namespace SendRecieveUDP.Service.Application
+{
+    public class ApplicationRunner
```
You split the functionality well, but the name should convey the job of this class just like any other service, ApplicationRunner iis too generic

nadav5 committed:
[REFACTOR] Improve naming and constants

Deleting an unnecessary constant and using an existing one.

Changing names to be more accurate
